### PR TITLE
fixes 10357: Update Partitionning Setting Flow for Profiler

### LINKED ISSUE #10357: Update Partitionning Setting Flow for Profiler
## Context
In https://github.com/open-metadata/OpenMetadata/pull/10350 we introduced `COLUMN-VALUE` and `INTEGER-RANGE` partitionning type with additional fields added `partitionValues`, `partitionIntegerRangeStart`, `partitionIntegerRangeEnd` to the `partitionProfilerConfig` property

```json
"partitionProfilerConfig": {
      "description": "This schema defines the partition configuration used by profiler.",
      "type": "object",
      "javaType": "org.openmetadata.schema.type.PartitionProfilerConfig",
      "properties": {
        "enablePartitioning": {
          "description": "whether to use partition",
          "type": "boolean",
          "default": false
        },
        "partitionColumnName": {
          "description": "name of the column to use for the partition",
          "type": "string"
        },
        "partitionIntervalType": {
          "description": "type of partition interval",
          "type": "string",
          "enum": [
            "TIME-UNIT",
            "INTEGER-RANGE",
            "INGESTION-TIME",
            "COLUMN-VALUE"
          ]
        },
        "partitionInterval": {
          "description": "The interval to use for the partitioning",
          "type": "integer"
        },
        "partitionIntervalUnit": {
          "description": "unit used for the partition interval",
          "type": "string",
          "enum": [
            "YEAR",
            "MONTH",
            "DAY",
            "HOUR"
          ]
        },
        "partitionValues": {
          "description": "unit used for the partition interval",
          "type": "array"
        },
        "partitionIntegerRangeStart": {
          "description": "start of the integer range for partitioning",
          "type": "integer",
          "default": null
        },
        "partitionIntegerRangeEnd": {
          "description": "end of the integer range for partitioning",
          "type": "integer",
          "default": null
        }
      }
    }
```

With the existing flow user can only choose partitionning for date/datetime/timestamp column and need to fill out all the field `Interval` and `Interval Unit` -- see screenshot below.

Screenshot 2023-02-28 at 8 31 14 AM

## Requirements
With the introduction of 2 additional partitionning type the workflow need to be modified to allow users to choose what partitionning field they want to use and display the right properties based on the interval type.

## Flow
At first only `Interval Type` and `Column Name` should be displayed - see screenshot below. The list of column displayed should depend on the `Interval Type`. It should be as follow

| Interval Type      | Displayed Column  Type |
| ----------- | ----------- |
| TIME-UNIT/INGESTION-TIME      | datetime/date/timestamp       |
| INTEGER-RANGE   | int/bigint        |
| COLUMN-VALUE   | varchar        |

Screenshot 2023-02-28 at 8 37 21 AM

Once an `Interval Type` has been selected the displayed options should be as followed

**TIME-UNIT/INGESTION-TIME**
Screenshot 2023-02-28 at 8 43 04 AM

**INTEGER-RANGE**
Screenshot 2023-02-28 at 8 48 14 AM

**COLUMN-VALUE**
Screenshot 2023-02-28 at 8 48 34 AM

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes:

Fixes #10357

Added partition for profiler modal based on partition type

image

image
image

image

https://user-images.githubusercontent.com/71748675/227720013-5f130f28-4d38-493f-b1a5-cc8dca0ca380.mov

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx [hunk 1]
```diff
@@ -193,7 +198,10 @@ const ProfilerSettingsModal: React.FC<ProfilerSettingsModalProps> = ({
         enablePartition: partitioning.enablePartitioning || false,
       });
 
-      form.setFieldsValue({ ...partitioning });
+      form.setFieldsValue({
+        ...partitioning,
+        partitionValues: partitioning.partitionValues || [''],
+      });
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:204]:
```suggestion
      form.setFieldValue(
        "partitionValues", partitioning.partitionValues || ['']
      );
```

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx [hunk 2]
```diff
@@ -121,9 +124,15 @@ const ProfilerSettingsModal: React.FC<ProfilerSettingsModalProps> = ({
     return metricsOptions;
   }, [columns]);
 
-  const { partitionColumnOptions, isPartitionDisabled } = useMemo(() => {
+  const partitionIntervalType = Form.useWatch(['partitionIntervalType'], form);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:127]: typeof `partitionIntervalType` can be inferred if we provide proper types to form

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx [hunk 3]
```diff
@@ -546,66 +584,26 @@ const ProfilerSettingsModal: React.FC<ProfilerSettingsModalProps> = ({
...
-                    disabled={!state?.enablePartition}
-                    options={partitionColumnOptions}
-                    placeholder={t('message.select-column-name')}
-                    size="middle"
-                  />
-                </Form.Item>
+              <Col span={24}>
+                <Space align="center" size={12}>
+                  <p>{t('label.enable-partition')}</p>
+                  <Form.Item className="m-b-0" name="enablePartitioning">
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:591]:
```suggestion
                  <Form.Item label={t('label.enable-partition')} className="m-b-0" name="enablePartitioning">
```

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx [hunk 4]
```diff
@@ -626,64 +624,236 @@ const ProfilerSettingsModal: React.FC<ProfilerSettingsModalProps> = ({
...
               <Col span={12}>
                 <Form.Item
                   className="m-b-0"
                   label={
-                    <span className="text-xs">{t('label.interval-unit')}</span>
+                    <span className="text-xs">
+                      {t('label.column-entity', {
+                        entity: t('label.name'),
+                      })}
+                    </span>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:635]: If this is what style we want to apply on every form label can we do it globally?

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx [hunk 5]
```diff
@@ -626,64 +624,236 @@ const ProfilerSettingsModal: React.FC<ProfilerSettingsModalProps> = ({
...
+                    options={partitionColumnOptions}
+                    placeholder={t('message.select-column-name')}
                     size="middle"
                   />
                 </Form.Item>
               </Col>
+              {[
+                PartitionIntervalType.IngestionTime,
+                PartitionIntervalType.TimeUnit,
+              ].includes(partitionIntervalType) ? (
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:663]: Batter to have it in constant

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:204]: there are multiple fields I am setting here but, from which `partitionValues` is the Form.List so if API don't have data, I am manually setting empty data so one filed is visible by default, without clicking add button

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx (hunk 4 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:635]: this is not the usual style, but it is particularly related to this form, to show the subsection

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx (hunk 3 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:592]: Our form has a layout type Vertical and we need a horizontal layout for the switch.

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx (hunk 1 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:205]: You can do it via initialValue then

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**